aruco_detect/src/aruco_detect.cpp

```diff
@@ -489,7 +490,7 @@ bool FiducialsNode::enableDetectionsCallback(std_srvs::SetBool::Request &req,
 }
 
 
-FiducialsNode::FiducialsNode() : nh(ros::NodeHandle("~")), it(nh)
```

Comment:
`nh` is a node handle in the default (global) namespace. If you publish a topic `foo` with `nh` it will map to topic `/foo`. pnh is a node handle that is in the `~` or private namespace. If you publish a topic `foo` with `pnh` it will map to `/node_name/foo`.

By publishing topics with `nh` and specifying no namespace at all (no leading slash) this makes it possible to launch multiple of this node using the namespace functionality in roslaunch.

Reply:
`Thanks for the explanation.  I'm confused because the call to the `NodeHandle` constructor is passed `"~"` for both `nh` and `pnh`.  This seems different to https://github.com/ros-planning/navigation/blob/d99320a5ff8c05f3a6246810b044ccbd00524107/move_base/src/move_base.cpp#L62 where it's clear that `nh` and `pnh` are in different namespaces.

Reply:
It is confusing because nh is default constructed, so there is no call to the constructor anywhere. I will add an explicit default constructor so that it is less confusing.

Reply:
Oh I see now, we are doing the publishing and subscribing with `nh` without the leading `/` and using `pnh` for params.  At first sight, I thought the change was just to rename `nh` to `pnh`